keyvalue/client-internal/src/main/java/tachyon/client/keyvalue/KeyValueSystem.java

```diff
@@ -38,14 +38,22 @@
     private Factory() {} // to prevent initialization
 
     /**
-     * @return a {@link KeyValueSystem} instance
+     * @return a (cached) {@link KeyValueStores} instance
      */
     public static synchronized KeyValueSystem create() {
       if (sKeyValueSystem == null) {
-        sKeyValueSystem = new BaseKeyValueSystem();
+        reset();
       }
       return sKeyValueSystem;
     }
+
+    /**
+     * {@link tachyon.client.ClientContext} may be reset in different tests running in the same JVM,
```

Comment:
this method seems test only. for those methods, a general rule right now in our code base is to avoid having them in the production code, but creating a testutil class in testing code and using whitebox to reset the static singleton object.